_posts/2016-02-24-session-user-flow.markdown

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title: "Use session variables to optimize your user flow"
+date: 2016-02-24 13:37:00
+categories: ["ruby", "rails", "sessions"]
+author: "schmierkov"
+---
+
+Sessions provide you a nice little data storage feature where the application does not need to get the informations directly from the database. So you do not have to persist data in your database and can easily store infos to a user on the fly. This is a nice way to enhance the user experience on your page. Let's say you want to show all users, from a specific area, a new fancy sign up form and the rest only sees the old form. With storing these location informations in a session variable, you do not need to persist this infos in your database.
+
+<!--more-->
+
+That can be pretty handy in many ways, if you think about changing the content of a website based on infos in that session. Handling a lot of session informations can be pretty overwhelming, because you have tons of crazy methods flying around. So my idea was once to stuff all these infos into an object. This way I could test it and do wild things with the data, but the frustration comes as soon as you change your Object stored in your session.
```

Comment:
"Handling a lot of session informations can be pretty overwhelming, because you have tons of crazy methods flying around." was my idea to describe the complexity of dealing with the session storage without objects that improve the working with it, I'm going to rephrase it, since it sounds confusing if I read it now :laughing:  thanks :+1: